clang/runtime/dpct-rt/include/dpl_extras/iterators.h.inc

```diff
@@ -325,6 +325,30 @@ public:
 };
 // DPCT_LABEL_END
 
+// DPCT_LABEL_BEGIN|io_iterator_pair|dpct
+// DPCT_DEPENDENCY_EMPTY
+// DPCT_CODE
+template <typename IterT> struct io_iterator_pair {
+  inline io_iterator_pair() : __selector(false) {}
+
+  inline io_iterator_pair(const IterT &first_input, const IterT &first_output)
+      : __selector(false) {
+    __iter[0] = first_input;
+    __iter[1] = first_output;
+  }
+
+  inline IterT input() const { return __selector ? __iter[1] : __iter[0]; }
+
+  inline IterT output() const { return __selector ? __iter[0] : __iter[1]; }
+
+  inline void swap() { __selector = !__selector; }
+
+  bool __selector;
+
+  IterT __iter[2];
```

Comment:
`__selector` and `__iter` are both public, is this intended? I would not expect public members to be prefixed by double underscores.

Reply:
Fixed naming convention, these are intended to be public.